### PR TITLE
natspubsub: replace ugorji/codec with encoding/gob

### PR DESCRIPTION
**Pull request by jba**
The codec package used to encode and decode messages is an uneccesary
dependency for pubsub, and takes up 11M in the binary (according to
https://github.com/jondot/goweight). Replace it with gob.

### CODE REVIEW

File: pubsub/natspubsub/nats.go [hunk 1]
```diff
@@ -175,28 +175,11 @@ type topic struct {
...
-	// driver.Message.Metadata type
-	dm := driver.Message{}
-	mh.MapType = reflect.TypeOf(dm.Metadata)
-}
-
 // We define our own version of message here for encoding that
-// only encodes Body and Metadata. Otherwise we would have to
-// add codec decorations to driver.Message.
+// only encodes Body and Metadata.
 type encMsg struct {
```
Comment (vangent) [inline pubsub/natspubsub/nats.go]: Is this used anymore?

**Comment by vangent** [comment]
`FAIL: dependencies changed; compare listdeps.sh output with alldeps`

File: pubsub/natspubsub/nats.go (hunk 1 above)
Comment (jba) [reply pubsub/natspubsub/nats.go]: No. Removed.

**Comment by jba** [comment]
PTAL.

**Comment by eliben** [comment]
Wow, nice find!

**Pull request merged**